frontend/src/metabase/query_builder/components/DatasetEditor/DatasetQueryEditor.unit.spec.tsx

```diff
@@ -0,0 +1,152 @@
+import { screen } from "@testing-library/react";
+import fetchMock from "fetch-mock";
+import _ from "underscore";
+
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { createMockEntitiesState } from "__support__/store";
+import { renderWithProviders } from "__support__/ui";
+import { Card } from "metabase-types/api";
+import {
+  createMockCard,
+  createMockCollection,
+  createMockNativeDatasetQuery,
+} from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import { createMockState } from "metabase-types/store/mocks";
+import { checkNotNull } from "metabase/core/utils/types";
+import { getMetadata } from "metabase/selectors/metadata";
+
+const { NativeQueryEditor } = jest.requireActual(
+  "metabase/query_builder/components/NativeQueryEditor",
+);
+
+const TEST_DB = createSampleDatabase();
+
+const TEST_NATIVE_CARD = createMockCard({
+  dataset_query: createMockNativeDatasetQuery({
+    type: "native",
+    database: TEST_DB.id,
+    native: {
+      query: "select * from orders",
+      "template-tags": undefined,
+    },
+  }),
+});
+
+const ROOT_COLLECTION = createMockCollection({ id: "root" });
+
+interface SetupOpts {
+  card?: Card;
+  height?: number;
+  isActive: boolean;
+  readOnly?: boolean;
+}
+
+const setup = async ({
+  card = TEST_NATIVE_CARD,
+  height = 300,
+  isActive,
+  readOnly = false,
+}: SetupOpts) => {
+  setupDatabasesEndpoints([TEST_DB]);
+
+  const storeInitialState = createMockState({
+    entities: createMockEntitiesState({
+      databases: [createSampleDatabase()],
+      questions: [card],
+    }),
+  });
+  const metadata = getMetadata(storeInitialState);
+  const question = checkNotNull(metadata.question(card.id));
+  const query = question.query();
+
+  const { default: DatasetQueryEditor } = await import(
+    "metabase/query_builder/components/DatasetEditor/DatasetQueryEditor"
+  );
+
+  const { rerender } = renderWithProviders(
+    <DatasetQueryEditor
+      isActive={isActive}
+      height={height}
+      query={query}
+      question={question}
+      readOnly={readOnly}
+      onResizeStop={_.noop}
+    />,
+  );
+
+  return { query, question, rerender };
+};
+
+describe("DatasetQueryEditor", () => {
+  beforeEach(() => {
+    fetchMock.get("path:/api/collection", [ROOT_COLLECTION]);
+    fetchMock.get("path:/api/native-query-snippet", () => []);
+
+    jest.unmock("metabase/query_builder/components/NativeQueryEditor");
```

Comment:
Could we maybe extract this into a function so we can put the async import with a comment about the mocked component there? And use this function in 2 places in the test suite.

Reply:
If not, that's ok

Reply:
a74fea711fc8910bf7739178f313d95d2baa7bac